src/sentry/tasks/files.py

```diff
@@ -45,3 +45,9 @@ def delete_unreferenced_blobs(blob_ids):
                 # Do nothing if the blob was deleted in another task, or
                 # if had another reference added concurrently.
                 pass
+            except DatabaseError:
+                # The database could have crashed or closed our connection.
+                # Try removing the file directly. We can recover
+                # from blobs with missing files, but files with no blob record
+                # become orphaned forever.
+                get_storage().delete(blob.path)
```

Comment:
If this task will automatically reap the postgres `FileBlob`, any reason we don't just rely on this as the primary path and remove the extra complexity here?

Reply:
Do you mean that we should delete the file and then remove the postgres row?

Reply:
Yeah, or if I'm understanding the description properly, is just deleting the file enough? Or are there reasons we'd prefer to do both at the same time?

Referring to 
> If the postgres record was not deleted, the unreferenced blob cleanup task will reap the postgres record and filestore object.

Reply:
We could delete the file first, and then delete the record. Right now the logic is to delete the record and then delete the file. Reversing the order of operations would be safer.